tests/manage/storageclass/test_custom_storageclass_names.py

```diff
@@ -0,0 +1,141 @@
+from ocs_ci.ocs.resources.storage_cluster import (
+    patch_storage_cluster_for_custom_storage_class,
+    check_custom_storageclass_presence,
+)
+from ocs_ci.ocs.constants import (
+    OCS_COMPONENTS_MAP,
+)
+from fauxfactory import gen_alpha, gen_special
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@skipif_external_mode
+@skipif_ocs_version("<4.14")
+class TestCustomStorageClassNames:
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        """
+        Fixture to restore custom storage class names after testing.
+        """
+
+        def restore_custom_storage_class_names():
+            if config.ENV_DATA.get("custom_default_storageclass_names"):
+                for sc_type, sc_name in config.ENV_DATA.get(
+                    "storageclassnames", {}
+                ).items():
+                    try:
+                        patch_storage_cluster_for_custom_storage_class(
+                            sc_type, storage_class_name=sc_name
+                        )
+                    except Exception as e:
+                        log.info(f"Patch failed with an error: {e}")
+
+        request.addfinalizer(restore_custom_storage_class_names)
+
+    @pytest.mark.polarion_id("OCS-5148")
+    def test_custom_storageclass_post_deployment(self, request):
+        """
+        Test custom storage class creation post deployment.
+
+        Steps:
+            1. Ensure Storagecluster is in ready state.
+            2. Edit the storage cluster spec and add custom storage class names.
+            3. Verify that the storage class has been created as per the updates mentioned in the spec.
+            4. Remove the custom storage class names from the storage cluster spec.
+            5. Delete the storage class mentioned in the storage cluster spec.
+
+        """
+        custom_sc_list = []
+        sc_type_list = [
+            OCS_COMPONENTS_MAP["cephfs"],
+            OCS_COMPONENTS_MAP["rgw"],
+            OCS_COMPONENTS_MAP["blockpools"],
+        ]
+        for sc_type in sc_type_list:
+            random_sc_name = f"custom-{sc_type}-{gen_alpha()}".lower()
+            log.info(
+                f"Adding custom storageclass '{random_sc_name}' of type '{sc_type}' in storagecluster spec."
+            )
+            assert patch_storage_cluster_for_custom_storage_class(
+                sc_type, storage_class_name=random_sc_name
+            ), f"Failed to add custom storageclass '{random_sc_name}' of type '{sc_type}' in storagecluster spec."
+            custom_sc_list.append(random_sc_name)
+
+        assert (
+            check_custom_storageclass_presence()
+        ), "Error validating the created storage classes."
+
+        def teardown():
+            for sc_type in sc_type_list:
+                patch_storage_cluster_for_custom_storage_class(sc_type, action="remove")
+
+            for sc in custom_sc_list:
+                run_cmd(f"oc delete sc {sc}")
+
+        request.addfinalizer(teardown)
```

Comment:
should not be part of the test function but as a fixture

Reply:
This code has moved in fixture.